src/py/flwr/server/superlink/driver/driver_servicer.py

```diff
@@ -208,13 +204,47 @@ def PullServerAppInputs(
         self, request: PullServerAppInputsRequest, context: grpc.ServicerContext
     ) -> PullServerAppInputsResponse:
         """Pull ServerApp process inputs."""
-        raise NotImplementedError()
+        log(DEBUG, "DriverServicer.PullServerAppInputs")
+        state = self.state_factory.state()
+        ffs = self.ffs_factory.ffs()
+
+        with self.lock:
+            if request.HasField("run_id"):
+                run_id: Optional[int] = request.run_id
+            else:
+                run_id = state.get_pending_run_id()
+            if run_id is None:
+                return PullServerAppInputsResponse()
+
```

Comment:
let's add a single line comment of what lines below do?

Reply:
Good idea!